### PR TITLE
feat: teacher signup page 4

### PR DESCRIPTION
**Pull request by chenjanet**
## Notion ticket link

[Teacher Signup - Page 4 - Set Password](https://www.notion.so/uwblueprintexecs/Teacher-Signup-Page-4-Set-Password-f597ccec8b4d43e4a5a331c3e1b85bfc)

## Implementation description
* TeacherSignupFour.tsx: created 4th page of teacher signup flow
* alert.ts: updated theming for `Alert` component to create a new variant that matches the styling [here](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=860%3A43313&t=XGffj9z3OmGeOU3q-0); also did a bit of refactoring and updated AddAdminModal.tsx accordingly
* icons.ts: added a filled checkmark circle icon to be used in password validation

## Steps to test
* Validate that the icons turn from red to green as each of the password requirements are filled, and back to red if they become unfulfilled
* Submit when the passwords don't match and verify that [this error message](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=860%3A43313&t=XGffj9z3OmGeOU3q-0) is displayed
* Submit when the passwords match but at least one requirement is unfulfilled, and verify that [this error message](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=860%3A43347&t=XGffj9z3OmGeOU3q-0) is displayed

## What should reviewers focus on?
Screenshots:
Screen Shot 2022-12-11 at 11 27 03 AM
Screen Shot 2022-12-11 at 11 27 23 AM
Screen Shot 2022-12-11 at 11 28 01 AM

[Figma design](https://www.figma.com/file/cRjJxm8087kkymfH4ieKau/Platform-Working-File?node-id=860%3A43282&t=XGffj9z3OmGeOU3q-0)

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by joyce-shi — CHANGES_REQUESTED** [review]
Overall, LGTM! Super cool PR 😎. 

Question though: should we be using `useForm` features here? I believe that there is built in functionality for validation, errors and inputs? Specifically, I noticed that the `register` and `errors` props that are passed to the page are unused.

Additionally, I'm a bit confused about how state is generally stored in the `TeacherSignup` flow. Are we passing state collected from every page forward so that we can submit on this last page? And if so, where do we do this? i.e should we be doing something like [this](https://react-hook-form.com/advanced-usage#WizardFormFunnel)? 

cc @carissa-tang

**Comment by carissa-tang** [comment]
> Overall, LGTM! Super cool PR 😎.
> 
> Question though: should we be using `useForm` features here? I believe that there is built in functionality for validation, errors and inputs? Specifically, I noticed that the `register` and `errors` props that are passed to the page are unused.
> 
> Additionally, I'm a bit confused about how state is generally stored in the `TeacherSignup` flow. Are we passing state collected from every page forward so that we can submit on this last page? And if so, where do we do this? i.e should we be doing something like [this](https://react-hook-form.com/advanced-usage#WizardFormFunnel)?
> 
> cc @carissa-tang

Yea, here's the doc on the built-in validation features https://react-hook-form.com/get-started#Applyvalidation not sure how restrictive it is tho.. I think if everything works as is we can leave it be. 

In terms of state, we are technically just working with one parent component `TeacherSignup`, which passes down the useForm functions so the child components can call them. To update a field, you can call `<input {...register('field')} />` or `setValue('field', 'value')` ([example in my pr](https://github.com/uwblueprint/jump-math/pull/88/files#diff-b5fb0cffdc6e0ba7a9bc3bcb3d3bf4ceeae975f2f09fdcb679fb35d91b3f6895R47)). You can query the form values using [getValues](https://react-hook-form.com/api/useform/getvalues) or [watch](https://react-hook-form.com/api/useform/watch). At the end, we can call `handleSubmit` which passes all the form data (in a param called `data`) and we can call our endpoints on `data`.

For this PR, I think you are just missing setting the value in `useForm`. To avoid refactoring, you can prob just call `setValue("password", password)`  before `handleSubmit`.

File: frontend/src/components/auth/TeacherSignup/TeacherSignupFour.tsx [hunk 1]
```diff
@@ -1,23 +1,157 @@
...
+  const [displayMatchError, setDisplayMatchError] = useState(false);
+  const [displayRequirementError, setDisplayRequirementError] = useState(false);
+  const [minNineChars, setMinNineChars] = useState(false);
+  const [hasUppercase, setHasUppercase] = useState(false);
+  const [hasLowercase, setHasLowercase] = useState(false);
+  const [hasSpecial, setHasSpecial] = useState(false);
+  const [hasNum, setHasNum] = useState(false);
   // TODO: handle submission of data to the backend
-  const onSubmit = handleSubmit((data) => console.log(data));
+  const onSubmit = () => {
```
Comment (BobShoaun) [inline frontend/src/components/auth/TeacherSignup/TeacherSignupFour.tsx:63]: where is the preventDefault?? rejected

**Review by BobShoaun — CHANGES_REQUESTED** [review]
lgtm (lets get this MONEYY)

**Review by joyce-shi — APPROVED** [review]
LGTM :D

**Pull request merged**